pkg/streamingpromql/functions/rate_increase.go

```diff
@@ -114,8 +114,8 @@ func histogramRate(isRate bool, step types.RangeVectorStepData, hHead []promql.H
 				return histogram.ErrHistogramsIncompatibleSchema
 			}
 
-			if p.H.Schema < delta.Schema {
-				delta = delta.CopyToSchema(p.H.Schema)
+			if p.H.Schema < desiredSchema {
+				desiredSchema = p.H.Schema
```

Comment:
do we need to copy delta to the desired schema at each counter reset (ie before adding the previous value)?

Reply:
No - `Add` handles the case where the histograms have different schemas and will return a histogram with the lower of the two schemas (ie. the lower resolution).

Reply:
Thanks, forgot about that.